client/src/main/java/io/pravega/client/stream/impl/ReaderGroupState.java

```diff
@@ -139,6 +140,20 @@ public String getScopedStreamName() {
     }
     
     @Synchronized
+    Map<Stream, Map<Segment, Long>> getPositions() {
+        Map<Stream, Map<Segment, Long>> result = new HashMap<>();
+        for (Entry<Segment, Long> entry : unassignedSegments.entrySet()) {
+            result.computeIfAbsent(entry.getKey().getStream(), s -> new HashMap<>()).put(entry.getKey(), entry.getValue());
```

Comment:
Q: Did not understand why we need to use computeIfAbsent, since all entries in unassignedSegments and assignedSegments have to be unique, if present it implies an error state.